shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -274,6 +277,20 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
 }
 
 - (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  NSRange currentRange = ((FlutterTextRange*)range).range;
+  if ( (currentRange.location > 0 && [text length] > 0 &&
+      [[NSCharacterSet whitespaceCharacterSet] characterIsMember:[text characterAtIndex:0]] &&
+      [[NSCharacterSet whitespaceCharacterSet] characterIsMember:[[self text] characterAtIndex:currentRange.location-1]] )) {
+    if ([self.text characterAtIndex:currentRange.location-2] == '.' ||
+        [self.text characterAtIndex:currentRange.location-2] == ' ') {
```

Comment:
If `currentRange.location` is less than 2, this will result in an out-of-range access.